remote/client.go

```diff
@@ -122,8 +124,14 @@ func NewClient(ctx context.Context, node *p2p.QriNode, pub event.Publisher) (c C
 		})
 	}
 
+	pro, err := node.Repo.Profile()
+	if err != nil {
+		log.Debug("cannot get profile from repo, need username for access control on the remote to function")
+	}
```

Comment:
this is delightful. I was initially going to ask for a `profile` argument to `NewClient` to preserve a scenario where a user may be offline, but want to use a remote in HTTP-only mode.  But looking at this constructor, almost all of it relies on the node _not_ being nil. So this whole thing would need to be rewritten.

I'm also noticing that I'm asking for you to add arguments to constructors a lot in review lately, and want to make a note of a possible future pattern instead: Having multiple constructors.

If we want this "HTTP only" mode, we could instead define a new constructor:

```go
func NewHTTPClient(r repo.Repo, pub event.Publisher) (*Client, error)
```

I want to point this pattern out so you're aware of it. We have a tendency to emphasize one constructor doing many things. This means sometimes arguments can be nil, and other times not so much, and it's hard to tell when it is and isn't ok to pass nil as args. Having additional constructor functions might help make this less painful